### PR TITLE
[Cleanup] Format and merge all cuda files and their wrappers.

### PR DESCRIPTION
**Pull request by gpzlx1**
*  Merging CSR and CSC formats. The CSR format is now just an alias for CSC.
*  Move all fused operator into cuda/fusion.
    *  random_walk
    * metapath_random_walk
    * slice_sampling
*  Rename the cuda operators. Each operator is now named to start with FORMAT.
*  Merge some files into one file.
    * columnwise_slicing.cu, rowwise_slicing.cu -> slicing.cu

### CODE REVIEW

File: src/graph_ops.cc [hunk 1]
```diff
@@ -1,55 +1,40 @@
...
+#include "cuda/fusion/slice_sampling.h"
 #include "cuda/graph_ops.h"
-#include "cuda/heterograph_ops.h"
-#include "cuda/random_walk.h"
+#include "cuda/tensor_ops.h"
 
 namespace gs {
 
-std::shared_ptr<COO> GraphCSC2COO(std::shared_ptr<CSC> csc) {
+std::shared_ptr<COO> GraphCSC2COO(std::shared_ptr<CSC> csc, bool CSC2COO) {
```
Comment (Liu-rj) [inline src/graph_ops.cc:10]: The `bool CSC2COO` parameter seems not so straightfoward for me. Maybe four operators `CSC2COO`, `CSR2COO`, `COO2CSC`, `COO2CSR` is better.

File: src/graph_ops.cc [hunk 2]
```diff
@@ -60,79 +45,64 @@ std::shared_ptr<CSC> GraphCOO2CSC(std::shared_ptr<COO> coo, int64_t num_cols) {
...
   if (csc->indptr.device().type() == torch::kCUDA) {
     torch::Tensor sub_indptr, sub_indices, select_index;
     std::tie(sub_indptr, sub_indices, select_index) =
-        impl::OnIndptrSlicingCUDA(csc->indptr, csc->indices, column_ids);
+        impl::OnIndicesSlicingCUDA(csc->indptr, csc->indices, node_ids);
     return {std::make_shared<CSC>(CSC{sub_indptr, sub_indices, torch::nullopt}),
             select_index};
   } else {
-    LOG(FATAL) << "Not implemented warning";
+    std::cerr << "Not implemented warning";
```
Comment (Liu-rj) [inline src/graph_ops.cc:57]: Why not use `LOG(FATAL)` here?

File: src/graph_ops.cc (hunk 1 above)
Comment (gpzlx1) [reply src/graph_ops.cc:10]: Similar with parameter `reverse` in DGL. If CSC2COO is True, it turn csc to coo. If not, it turn csr to coo

**Review by Liu-rj — APPROVED** [review]
Looks nice.

**Pull request merged**